### PR TITLE
Add lint rule to prevent logging objects

### LINKED ISSUE #6512: Add lint rule to prevent logging objects
In https://github.com/mozilla/addons-frontend/issues/6369 we got DOS'd by our own log message that was logging a large object, thus allocating a lot of memory. Can we add a lint rule to prevent logging objects? Even if it were naive and just prevented passing non-string arguments to `log.info()`, that might be good enough. Lint rules can always be disabled -- it will at least make the developer think about the implications.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #6512

---

The  lint rule has been added to https://github.com/mozilla/eslint-plugin-amo#only-log-strings. I reviewed each error and tried to fix it in an appropriate manner. We can probably change some fixes. Let's discuss them here.

### CODE REVIEW

File: src/amo/pages/UserProfile/index.js [hunk 1]
```diff
@@ -277,7 +277,7 @@ export class UserProfileBase extends React.Component<InternalProps> {
 
     let errorMessage;
     if (errorHandler.hasError()) {
-      log.warn('Captured API Error:', errorHandler.capturedError);
+      log.warn(`Captured API Error: ${errorHandler.capturedError}`);
```
Comment (willdurand) [inline src/amo/pages/UserProfile/index.js]: I took this approach but maybe we want to log the full `capturedError`?

File: src/core/addonManager.js [hunk 2]
```diff
@@ -92,7 +92,9 @@ export function getAddon(
       if (!addon) {
         throw new Error('Addon not found');
       }
-      log.info('Add-on found', addon);
+      // eslint-disable-next-line amo/only-log-strings
+      log.debug('Add-on found', addon);
```
Comment (willdurand) [inline src/core/addonManager.js]: The more I think about the ESLint rule, the more I believe `debug` should not be taken into account. WDYT?

File: src/core/middleware/prefixMiddleware.js [hunk 3]
```diff
@@ -13,7 +13,6 @@ import log from 'core/logger';
 export function prefixMiddleware(req, res, next, { _config = config } = {}) {
   // Split on slashes after removing the leading slash.
   const URLParts = req.originalUrl.replace(/^\//, '').split('/');
-  log.debug(URLParts);
```
Comment (kumar303) [inline src/core/middleware/prefixMiddleware.js]: 🤡 🔪

File: src/core/reducers/errors.js [hunk 4]
```diff
@@ -22,6 +22,7 @@ function getMessagesFromError(error) {
     code: ERROR_UNKNOWN,
     messages: [],
   };
+  // eslint-disable-next-line amo/only-log-strings
   log.info('Extracting messages from error object:', error);
```
Comment (kumar303) [inline src/core/reducers/errors.js]: This should probably be `debug`

File: src/core/tracking.js [hunk 5]
```diff
@@ -167,6 +168,7 @@ export class Tracking {
 
   log(...args) {
     if (this._log) {
+      // eslint-disable-next-line amo/only-log-strings
```
Comment (kumar303) [inline src/core/tracking.js]: Why isn't `this.log` flagged by the lint rule? I think this log statement should have been flagged:

https://github.com/mozilla/addons-frontend/blob/48b2bccc4eedf79d945f1fa310a0c3ed8a7c42a6/src/core/tracking.js#L248

That one should probably change to `debug` anyway.

File: src/core/utils/compatibility.js [hunk 6]
```diff
@@ -61,7 +61,9 @@ export function getCompatibleVersions({
     }
 
     if (!supportsClientApp) {
-      _log.warn('addon is incompatible with clientApp', { addon, clientApp });
+      _log.warn(
+        `addon "${addon.guid}" is incompatible with clientApp: "${clientApp}"`,
```
Comment (kumar303) [inline src/core/utils/compatibility.js]: Maybe prefix this with `guid:"${addon.guid}"` so it's unambiguous

File: src/core/tracking.js (hunk 5 above)
Comment (willdurand) [reply src/core/tracking.js]: The rule does not have `this` in the list of objects and not `log` in the list of methods.

File: src/core/utils/compatibility.js (hunk 6 above)
Comment (willdurand) [reply src/core/utils/compatibility.js]: yup

File: src/amo/components/Search/index.js [hunk 7]
```diff
@@ -198,7 +198,7 @@ export class SearchBase extends React.Component<InternalProps> {
     } = this.props;
 
     if (errorHandler.hasError()) {
-      log.warn('Captured API Error:', errorHandler.capturedError);
+      log.warn(`Captured API Error: ${errorHandler.capturedError}`);
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/amo/components/Search/index.js]: > So how about `${errorHandler.capturedError.messages}`?

wfm!

File: src/core/tracking.js (hunk 5 above)
Comment (kumar303) [reply src/core/tracking.js]: oh, I see. It's because it walks an AST.

File: src/core/reducers/errors.js [hunk 8]
```diff
@@ -69,7 +69,7 @@ function getMessagesFromError(error) {
   }
 
   if (!errorData.messages.length) {
-    log.warn('Error object did not contain any messages', error);
+    log.warn(`Error object did not contain any messages: ${error}`);
```
Comment (kumar303) [inline src/core/reducers/errors.js]: I guess the intention of logging this was to see the full data structure of `error`, to see what `error.response.data` looks like (if it exists). But, I dunno, I guess it is fine how you have it. We can always revisit it if we need to debug a related problem.

**Pull request merged**